### PR TITLE
refactor follow context

### PR DESCRIPTION
**Pull request by hsnice16**
### This PR will add refactored code for the follow context

In this PR, **153 additions and 165 deletions** of lines of code
- many files contain only 5 to 6 lines of code

### Please review this PR and give your feedback

---

### CODE REVIEW

File: src/custom-hooks/useBroadcastDetails.js [hunk 1]
```diff
@@ -115,8 +118,14 @@ export const useBroadcastDetails = ({ _id, username, createdAt, likedBy }) => {
           ).map((option) => ({
             ...option,
             handleClick: isInLoggedUserFollowing
-              ? () => postUnfollowCall(username)
-              : () => postFollowCall(username),
+              ? () => {
+                  dispatch(setFollowUsername(username));
+                  postUnfollowCall(username);
```
Comment (ajay-mandaviya) [inline src/custom-hooks/useBroadcastDetails.js:123]: avoid this type of nesting function calling better to use if else

**Review by ajay-mandaviya — APPROVED** [review]

File: src/redux/api/userAPI.js [hunk 2]
```diff
@@ -48,12 +48,36 @@ export const userAPI = baseAPI.injectEndpoints({
...
+      query: (username) => ({
+        body: {},
+        method: "POST",
+        url: `/users/follow/${username}`,
+        headers: {
+          authorization: authCookieHandler().getCookies().userAuthToken,
+        },
+      }),
+    }),
+
```
Comment (rahulyadav139) [inline src/redux/api/userAPI.js:62]: Appreciation: a great way to save all user API in one place. Great work.

**Review by rahulyadav139 — APPROVED** [review]

**Review by pradipgharge — APPROVED** [review]
This is a small Pr, looks good to merge

File: src/custom-hooks/useBroadcastDetails.js (hunk 1 above)
Comment (hsnice16) [reply src/custom-hooks/useBroadcastDetails.js:123]: Thanks for suggesting, Ajay
Actually, it was only one level nesting that's why I thought to use the ternary operator here.

**Comment by hsnice16** [comment]
Thank you for reviewing

**Pull request merged**